pkg/assetattributor/asset_inventory_api_component.go

```diff
@@ -0,0 +1,34 @@
+package assetattributor
+
+import (
+	"context"
+)
+
+// AssetInventoryAPIAttributorConfig defines the configuration options for a AssetInventoryAPIAttributor.
+type AssetInventoryAPIAttributorConfig struct {
+	Host           string `description:"The hostname of the asset-inventory-api service to query."`
+	CloudAssetPath string `description:"The cloud asset URI path for asset-inventory-api"`
+}
+
+// Name is used by the settings library to replace the default naming convention.
+func (c *AssetInventoryAPIAttributorConfig) Name() string {
+	return "assetinventoryapiattributor"
```

Comment:
nit: Why not `attributor` or something else shorter?

Reply:
This is intended to differentiate it from other potential `AssetAttributor` implementations and signify that it is intended to query an `asset-inventory-api` service.

Reply:
I think the better way to handle that is with an aggregate component rather than over specifying the individual one. For example,

```golang
type AttributorConfig struct {
  Type string `description:"Selection of attributor implementations."`
  // The aggregate component contains the other configs so that
  // it automatically parses any set values.
  AssetInventory *AssetInventoryConfig
}
type AttributorComponent struct{
  // The component also contains each of the other components so it can use their methods later.
   AssetInventory *AssetInventoryComponent
}
func (c *AttributorComponent) Settings() *AttributorConfig {
  return &AttributorConfig{
    Type: "assetinventory",
    // generate the default settings for this component directly from it.
    AssetInventory: c.AssetInventory.Settings(),
  }
}

func (c *AttributorComponent) New(ctx context.Context, conf *AttributorConfig) (domain.Attributor, error) {
  switch {
  strings.EqualFold(conf.Type, "assetinventory"):
    // all of these components should share the output types so we can delegate all of the construction to the other component here and return whatever it wants.
    return c.AssetInventory.New(ctx, conf.AssetInventory)
  default:
    return nil, fmt.Errorf("unknown attributor type %s", conf.Type)
  }
}
```

To configure this you'd then have something like:

```bash
NEXPOSE_ATTRIBUTOR_TYPE="assetinventory"
NEXPOSE_ATTRIBUTOR_ASSETINVENTORY_HOST="https://localhost"
```

The aggregate component structure I recommend for places where we will have multiple implementations of a thing. The naming is pretty fluid. I mostly just didn't like the long var name because I think it's hard to read when all lower or upper case.

Reply:
I've changed the name to `attributor`